packages/fx-core/src/plugins/resource/bot/plugin.ts

```diff
@@ -75,7 +75,7 @@ export class TeamsBotImpl {
     await this.config.restoreConfigFromContext(context);
     Logger.info(Messages.PreScaffoldingBot);
 
-    const rawWay = this.ctx.answers?.get(QuestionNames.WAY_TO_REGISTER_BOT);
+    const rawWay = this.ctx.answers![QuestionNames.WAY_TO_REGISTER_BOT];
```

Comment:
it's risky from ? to ! since answers is not controlled by the bot plugin.
To be robust, do some precondition checking?

Reply:
It's a workaround, because the answer is definitely existing

Reply:
In the future, this inputs will be as a parameter of life cycle API:
![image](https://user-images.githubusercontent.com/1658418/119772142-4d682a00-bef1-11eb-88d2-02fb4a0638f1.png)

Reply:
I am a little worried that this PR is a workaround or temporary solution, and after this big refactor we need another several PR to improve code quality. Who will do this kind of following code refactor and if we forget to do that if will decrease the code quality of the whole product.